### PR TITLE
feat: [PIE-4739]: Allowed Values POC

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 14 Issues 
* Code issue 14 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10421)

File: src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx [hunk 1]
```diff
@@ -6,25 +6,106 @@
...
+const TYPE_TO_VALIDATOR = {
+  [OPTIONS_TYPE.TIME]: (getString: (key: StringKeys, vars?: Record<string, any>) => string) =>
+    Yup.object().shape({
+      timeout: getDurationValidationSchema({ minimum: '10s' }).required(getString('validation.timeout10SecMinimum'))
+    })
+}
+
+const renderField = ({ fetchValues, getString, options, optionsType, isReadonly, formik }: RenderFieldProps) => {
+  const { setErrors, errors, setFieldTouched, setFieldValue } = formik
+  if (fetchValues) {
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx:51]: Looks like fetchValues is used to show Dropdown and we do have plan to integrate Select component as well. So, is it possible to pass proper optionsType for Select and get rid of fetchValues as prop name is ambiguous as well.

File: src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx [hunk 2]
```diff
@@ -6,25 +6,106 @@
...
+              const err = yupToFormErrors<{ timeout: string }>(e)
+              setErrors({ ...errors, allowedValues: err.timeout })
+            }
+          }
+        }
+      }
+    }
+  }
+
+  return (
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx:95]: Is it possible to add **KVTagInput** as part of default block of switch...case ?

File: src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx:51]: Definitely , we will iterate on it, once we start on Select component

File: src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx (hunk 2 above)
Comment (ishantmehtaharness) [reply src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx:95]: Definitely, we will iterate on it, once we find that requirement

Comment (ishantmehtaharness) [reply src/modules/10-common/components/ConfigureOptions/AllowedValuesField.tsx:95]: should be a small change

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/HarnessApprovalStepMode.tsx [hunk 3]
```diff
@@ -86,6 +86,7 @@ function FormContent({
           <ConfigureOptions
             value={formik.values.timeout || ''}
             type="String"
+            // allowedValuesType={ALLOWED_VALUES_TYPE.TIME}
```
Comment (vkbansal-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/HarnessApprovalStepMode.tsx:89]: do not commit this

Comment (ishantmehtaharness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/HarnessApprovalStepMode.tsx:89]: Will remove in follow-up PR.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/HarnessApprovalStepMode.tsx (hunk 3 above)
Comment (ishantmehtaharness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/HarnessApprovalStepMode.tsx:89]: Removed here
https://github.com/harness/harness-core-ui/pull/10573/files

**Pull request merged**